### PR TITLE
feature: fields custom components

### PR DESCRIPTION
**Pull request by Paul-Bob**

### CODE REVIEW

File: docs/3.0/field-options.md [hunk 1]
```diff
@@ -327,3 +327,48 @@ Now, all fields will have the stacked layout throughout your app.
...
+To start customizing the field components, you can eject one or multiple field components using the `avo:eject` command. Ejecting a field component generates the necessary files for customization. Here's how you can use the `avo:eject` command:
+
+#### Ejecting All Components for a Field
+`$ rails g avo:eject --field-components <field_type>`
+
+Replace `<field_type>` with the desired field type. For instance, to eject components for a Trix field, use:
+
+`$ rails g avo:eject --field-components trix`
+
+This command will generate the files for both the edit and show components of the Trix field, for each field type the amount of components may vary.
```
Comment (adrianthedev) [inline docs/3.0/field-options.md:344]: Let's provide an example where we eject the `index` view too. I immediately asked myself "why don't they eject the index view as well"?
It took a minute to click for me that trix does not have an Index field.

File: docs/3.0/field-options.md [hunk 2]
```diff
@@ -327,3 +327,48 @@ Now, all fields will have the stacked layout throughout your app.
...
+```ruby
+field :description,
+  as: :trix,
+  attachment_key: :description_file,
+  visible: -> { resource.params[:show_native_fields].blank? },
+  components: -> do
+    {
+      field_show_component: Avo::Fields::Admin::TrixField::ShowComponent
+    }
+  end
```
Comment (adrianthedev) [inline docs/3.0/field-options.md:365]: let's show the Hash example first and tell them there's a block option too.

File: docs/3.0/customization.md [hunk 3]
```diff
@@ -228,20 +228,52 @@ Have the same output:
 create  app/components/avo/index/table_row_component.rb
 create  app/components/avo/index/table_row_component.html.erb
 ```
+:::
+
+:::option `--field-components`
```
Comment (adrianthedev) [inline docs/3.0/customization.md:233]: Let's link to the field-options path somewhere below this so they can reach it more easily.

File: docs/3.0/field-options.md [hunk 4]
```diff
@@ -327,3 +327,48 @@ Now, all fields will have the stacked layout throughout your app.
...
 :::
+
+:::option `components`
+The field's `components` option allows you to customize the view components used for rendering the field in all, `index`, `show` and `edit` views. This provides you with a high degree of flexibility.
+
+### Ejecting the field components
+To start customizing the field components, you can eject one or multiple field components using the `avo:eject` command. Ejecting a field component generates the necessary files for customization. Here's how you can use the `avo:eject` command:
+
+#### Ejecting All Components for a Field
+`$ rails g avo:eject --field-components <field_type>`
```
Comment (adrianthedev) [inline docs/3.0/field-options.md:338]: Also, let's link to the eject command section for more details.

"For more advanced usage, see the `--field-components` page."

**Review by adrianthedev — APPROVED** [review]
Very nice! Left a few comments.

Thank you!

**Comment by Paul-Bob** [comment]
@adrianthedev review again please since last commit has a bunch of changes according to your last review

**Review by adrianthedev — APPROVED** [review]
Very well written!
Thank you!

**Pull request merged**